### PR TITLE
Add Compression Test to ensure handling large outputs during the flush stage

### PR DESCRIPTION
**Pull request by stwrt**
Ensure compression streams handles large flush output during the closing stages of the compression stream.

### CODE REVIEW

**Review by annevk — APPROVED** [review]
It's interesting that inflation doesn't need the format. I'd slightly prefer asserting the output of compression, but I guess other tests cover that already.

File: compression/compression-large-flush-output.any.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+async function compressData(chunk, format) {
+  const cs = new CompressionStream(format);
+  const writer = cs.writable.getWriter();
+ 
+  writer.write(chunk);
+
+  const closePromise = writer.close();
+  const out = [];
+  const reader = cs.readable.getReader();
```
Comment (ricea) [inline compression/compression-large-flush-output.any.js:18]: Is there a reason you can't use concatenateStream from resources/concatenate-stream.js ?

Comment (stwrt) [reply compression/compression-large-flush-output.any.js:18]: Let me a take a look. This was written outside the whole WPT framework, so there may be a few more things we could simplify here.

File: compression/compression-large-flush-output.any.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+const compressedData = await compressData(data, 'gzip');
+// decompress with pako, and check that we got the same result as our original string
+assert_array_equals(expectedValue, pako.inflate(compressedData), 'value should match');
+}, `gzip compression with large flush output`);
+
+promise_test(async t => {
+const compressedData = await compressData(data, 'deflate-raw');
+// decompress with pako, and check that we got the same result as our original string
+assert_array_equals(expectedValue, pako.inflateRaw(compressedData), 'value should match');
+}, `deflate-raw compression with large flush output`);
```
Comment (nt1m) [inline compression/compression-large-flush-output.any.js:58]: Small nit, but can you indent the code inside each promise_test ?

File: compression/compression-large-flush-output.any.js (hunk 1 above)
Comment (stwrt) [reply compression/compression-large-flush-output.any.js]: Fixed

File: compression/compression-large-flush-output.any.js (hunk 2 above)
Comment (stwrt) [reply compression/compression-large-flush-output.any.js]: Done

**Review by ricea — APPROVED** [review]
lgtm

**Pull request merged**